<h2>Josh Vrylle Castillo</h2>
<h3>Data Analysis on Accidents</h3>

<h3>Installing and Importing Modules</h3>

In [1]:
pip install numpy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
pip install seaborn

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [4]:
pip install matplotlib


[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [5]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.0 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [6]:
import numpy as np
import pandas as pd
import seaborn as sns
import scipy as sp
import matplotlib.pyplot as plt

In [70]:
from scipy.stats import f_oneway

In [7]:
accident = pd.read_csv('dataset\\accident_data.csv')

In [8]:
accident.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [9]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

<h3>Cleaning the Null Values</h3>

In [10]:
accident.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

In [11]:
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna('Unaccounted Road Conditions')
accident['Road_Type'] = accident['Road_Type'].fillna('Unaccounted Road Type')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna('Unaccounted')
accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna('Unaccounted Weather Cobditions')
accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])

In [12]:
# accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].fillna(accident['Road_Surface_Conditions'].mode()[0])
# accident['Road_Type'] = accident['Road_Type'].fillna(accident['Road_Type'].mode()[0])
# accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].fillna(accident['Urban_or_Rural_Area'].mode()[0])
# accident['Weather_Conditions'] = accident['Weather_Conditions'].fillna(accident['Weather_Conditions'].mode()[0])
# accident['Latitude'] = accident['Latitude'].fillna(accident['Latitude'].mode()[0])
# accident['Longitude'] = accident['Longitude'].fillna(accident['Longitude'].mode()[0])

<h3>Cleaning the Data Types</h3>

In [13]:
accident.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660679 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660679 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  660679 non-null  object 
 10  Road_Type                660679 non-null  object 
 11  Urban_or_Rural_Area      660679 non-null  object 
 12  Weather_Conditions       660679 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

In [14]:
accident['Accident_Severity'] = accident['Accident_Severity'].astype('category')
accident['Light_Conditions'] = accident['Light_Conditions'].astype('category')
accident['District Area'] = accident['District Area'].astype('category')
accident['Road_Surface_Conditions'] = accident['Road_Surface_Conditions'].astype('category')
accident['Road_Type'] = accident['Road_Type'].astype('category')
accident['Urban_or_Rural_Area'] = accident['Urban_or_Rural_Area'].astype('category')
accident['Weather_Conditions'] = accident['Weather_Conditions'].astype('category')
accident['Vehicle_Type'] = accident['Vehicle_Type'].astype('category')
accident['Latitude'] = accident['Latitude'].astype('category')
accident['Longitude'] = accident['Longitude'].astype('category')

<h3>Modifying the Date</h3>

In [15]:
accident['Accident Date'] = accident['Accident Date'].str.strip()

accident['Accident Date'] = accident['Accident Date'].str.replace('/','-')

In [16]:
accident['Accident Date'] = pd.to_datetime(accident['Accident Date'], dayfirst=True, errors = 'coerce')

In [17]:
accident['Year'] = accident['Accident Date'].dt.year.astype('category')
accident['Month'] = accident['Accident Date'].dt.month.astype('category')
accident['Day'] = accident['Accident Date'].dt.day.astype('category')
accident['DaysOfWeek'] = accident['Accident Date'].dt.dayofweek

accident['DaysOfWeek'] = accident['DaysOfWeek'].map({0:'Monday',1:'Tuesday',2:'Wednesday',3:'Thursday',4:'Friday',5:'Saturday',6:'Sunday'})
accident['Month'] = accident['Month'].map({1:'January',2:'February',3:'March',4:'April',5:'May',6:'June',7:'July',8:'August',9:'September',10:'October',11:'November',12:'December'})

In [18]:
accident.isnull().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
Year                       0
Month                      0
Day                        0
DaysOfWeek                 0
dtype: int64

In [19]:
accident_2019 = accident[accident['Year'] == 2019]
accident_2020 = accident[accident['Year'] == 2020]
accident_2021 = accident[accident['Year'] == 2021]
accident_2022 = accident[accident['Year'] == 2022]

In [20]:
accident_january = accident[accident['Month'] == 'January']
accident_february = accident[accident['Month'] == 'February']
accident_march = accident[accident['Month'] == 'March']
accident_april = accident[accident['Month'] == 'April']
accident_may = accident[accident['Month'] == 'May']
accident_june = accident[accident['Month'] == 'June']
accident_july = accident[accident['Month'] == 'July']
accident_august = accident[accident['Month'] == 'August']
accident_september = accident[accident['Month'] == 'September']
accident_october = accident[accident['Month'] == 'October']
accident_november = accident[accident['Month'] == 'November']
accident_december = accident[accident['Month'] == 'December']

<h1>Valuable Insights</h1>

<h2>Question: Which factors contributes to the most Slight accidents? ( Light Conditions, Road Surface Conditions, Weather Conditions )</h2>

In [21]:
accident.groupby(['Accident_Severity','Light_Conditions'],observed=False).size().unstack()

Light_Conditions   Darkness - lighting unknown  Darkness - lights lit  \
Accident_Severity                                                       
Fatal                                       68                   1860   
Serious                                    794                  19130   
Slight                                    5622                 108345   

Light_Conditions   Darkness - lights unlit  Darkness - no lighting  Daylight  
Accident_Severity                                                             
Fatal                                   45                    1612      5076  
Serious                                360                    7174     60759  
Slight                                2138                   28651    419045

In [23]:
slight = accident[accident['Accident_Severity'] == 'Slight'].value_counts().sum()
slight

563801

In [24]:
accident[(accident['Accident_Severity'] == 'Slight')&(accident['Light_Conditions'] == 'Daylight')].value_counts().sum()


419045

In [41]:
419045 / slight

0.743249834604763

<h3>Insight 1: Daylight is the leading light condition that causes the most slight accidents from 2019 to 2022. 74.32% of the Slight accidents comes from Daylight condition.</h3>

In [26]:
accident.groupby(['Accident_Severity','Road_Surface_Conditions'],observed=False).size().unstack()

Road_Surface_Conditions     Dry  Flood over 3cm. deep  Frost or ice  Snow  \
Accident_Severity                                                           
Fatal                      5788                    23           193    35   
Serious                   61638                   152          2007   565   
Slight                   380395                   842         16317  5290   

Road_Surface_Conditions  Unaccounted Road Conditions  Wet or damp  
Accident_Severity                                                  
Fatal                                              2         2620  
Serious                                           70        23785  
Slight                                           654       160303

In [27]:
accident[(accident['Accident_Severity'] == 'Slight')&(accident['Road_Surface_Conditions'] == 'Dry')].value_counts().sum()

380395

In [28]:
381049 / slight

0.6758572616933989

<h3>Insight 2: Dry is the leading road condition that causes the most slight accidents from 2019 to 2022. 67.59% of the Slight accidents comes from Dry road condition. </h3>

In [29]:
accident.groupby(['Accident_Severity','Weather_Conditions'],observed=False).size().unstack()

Weather_Conditions  Fine + high winds  Fine no high winds  Fog or mist  Other  \
Accident_Severity                                                               
Fatal                             175                7100           82    165   
Serious                          1245               72046          483   1801   
Slight                           7134              441739         2963  15184   

Weather_Conditions  Raining + high winds  Raining no high winds  \
Accident_Severity                                                 
Fatal                                145                    848   
Serious                             1261                   9468   
Slight                              8209                  69380   

Weather_Conditions  Snowing + high winds  Snowing no high winds  \
Accident_Severity                                                 
Fatal                                  3                     36   
Serious                              109                    565   
Slight                               773                   5637   

Weather_Conditions  Unaccounted Weather Cobditions  
Accident_Severity                                   
Fatal                                          107  
Serious                                       1239  
Slight                                       12782

In [30]:
accident[(accident['Accident_Severity'] == 'Slight')&(accident['Weather_Conditions'] == 'Fine no high winds')].value_counts().sum()

441739

In [31]:
454521 / slight

0.8061727453480927

<h3>Insight 3: Fine no high winds is the leading weather condition that causes the most slight accidents from 2019 to 2022. 80.61% of the Slight accidents comes from Fine no high winds weather condition.</h3>

<h3>Insight 4: According to the data above, the highest accident severity is slight, followed by serious and fatal, respectively. The most common conditions for the Slight accidents are the following: 74.32% for Daylight, 67.58% for Dry roads, and 80.61% for Fine no high winds weather. This means that you are most prone to slight accidents while in the most ideal weather(fine no high winds), road condition(dry), and light condition(daylight).</h3>

<h2>Question: From the data gathered above. Slight accidents happen during the most ideal weather conditions. Does the same concept happen to Serious accidents? What about Fatal accidents? </h2>

In [32]:
accident.groupby(['Accident_Severity','Light_Conditions'],observed=False).size().unstack()

Light_Conditions   Darkness - lighting unknown  Darkness - lights lit  \
Accident_Severity                                                       
Fatal                                       68                   1860   
Serious                                    794                  19130   
Slight                                    5622                 108345   

Light_Conditions   Darkness - lights unlit  Darkness - no lighting  Daylight  
Accident_Severity                                                             
Fatal                                   45                    1612      5076  
Serious                                360                    7174     60759  
Slight                                2138                   28651    419045

In [33]:
accident.groupby(['Accident_Severity','Road_Surface_Conditions'],observed=False).size().unstack()

Road_Surface_Conditions     Dry  Flood over 3cm. deep  Frost or ice  Snow  \
Accident_Severity                                                           
Fatal                      5788                    23           193    35   
Serious                   61638                   152          2007   565   
Slight                   380395                   842         16317  5290   

Road_Surface_Conditions  Unaccounted Road Conditions  Wet or damp  
Accident_Severity                                                  
Fatal                                              2         2620  
Serious                                           70        23785  
Slight                                           654       160303

In [34]:
accident.groupby(['Accident_Severity','Weather_Conditions'],observed=False).size().unstack()

Weather_Conditions  Fine + high winds  Fine no high winds  Fog or mist  Other  \
Accident_Severity                                                               
Fatal                             175                7100           82    165   
Serious                          1245               72046          483   1801   
Slight                           7134              441739         2963  15184   

Weather_Conditions  Raining + high winds  Raining no high winds  \
Accident_Severity                                                 
Fatal                                145                    848   
Serious                             1261                   9468   
Slight                              8209                  69380   

Weather_Conditions  Snowing + high winds  Snowing no high winds  \
Accident_Severity                                                 
Fatal                                  3                     36   
Serious                              109                    565   
Slight                               773                   5637   

Weather_Conditions  Unaccounted Weather Cobditions  
Accident_Severity                                   
Fatal                                          107  
Serious                                       1239  
Slight                                       12782

<h3>Insight 5: The conditions that causes the most Serious accidents are the following: daylight(light condition), dry(road condition), and Fine no winds(weather condition)</h3>

<h3>Insight 6: The conditions that causes the most Fatal accidents are the following: daylight(light condition), dry(road condition), and Fine no winds(weather condition)</h3>

<h3>Insight 7: Answering the previous question. Surprisingly, the same concept applies to Serious and Fatal accidents. We can conclude that the 'ideal conditions' can cause the most accidents. We all know that in unideal conditions such as having fogs, snows, wet roads, and very dim roads, it is very dangerous to go outside. One possible reason why more accidents happen during ideal conditions is because there are more people outside, thus higher probability of them being in an accident. When an unideal condition is present like a typhoon, people tends to stay indoors, making them safer.</h3>

In [35]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious    2019-08-26  51.517715   
3       200701BS66128           Serious    2019-08-16  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    2022-02-18  57.374005   
660675  201091NM01881            Slight    2022-02-21  57.232273   
660676  201091NM01935            Slight    2022-02-23  57.585044   
660677  201091NM01964           Serious    2022-02-23  57.214898   
660678  201091NM02142           Serious    2022-02-28  57.575210   

                   Light_Conditions           District Area Longitude  \
0             Darkness - lights lit  Kensington and Chelsea -0.209082   
1                          Daylight  Kensington and Chelsea -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea -0.210215   
3                          Daylight  Kensington and Chelsea -0.202731   
4             Darkness - lights lit  Kensington and Chelsea -0.192487   
...                             ...                     ...       ...   
660674                     Daylight                Highland -3.467828   
660675       Darkness - no lighting                Highland -3.809281   
660676                     Daylight                Highland -3.862727   
660677       Darkness - no lighting                Highland -3.823997   
660678                     Daylight                Highland -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                    Road_Type Urban_or_Rural_Area  \
0          Single carriageway               Urban   
1          Single carriageway               Urban   
2       Unaccounted Road Type               Urban   
3          Single carriageway               Urban   
4       Unaccounted Road Type               Urban   
...                       ...                 ...   
660674     Single carriageway               Rural   
660675     Single carriageway               Rural   
660676     Single carriageway               Rural   
660677     Single carriageway               Rural   
660678       Dual carriageway               Rural   

                    Weather_Conditions                          Vehicle_Type  \
0                   Fine no high winds                                   Car   
1                Raining no high winds                                   Car   
2       Unaccounted Weather Cobditions                 Taxi/Private hire car   
3                   Fine no high winds  Bus or coach (17 or more pass seats)   
4       Unaccounted Weather Cobditions                         Other vehicle   
...                                ...                                   ...   
660674              Fine no high winds                                   Car   
660675              Fine no high winds                                   Car   
660676              Fine no high winds              

<h2>Question: Since we found out that ideal conditions makes you more prone to accidents. Where does all these accidents usually occur despite the influence of other conditions? Find out whether there are more accidents on Urban or Rural areas. </h2>

In [36]:
urban = accident[accident['Urban_or_Rural_Area'] == 'Urban']
rural = accident[accident['Urban_or_Rural_Area'] == 'Rural']

In [37]:
urban.value_counts().sum()

421663

In [38]:
rural.value_counts().sum()

238990

In [48]:
total_urban_rural = accident['Urban_or_Rural_Area'].value_counts().sum()
total_urban_rural

660679

In [49]:
421663/ total_urban_rural

0.6382267334060868

In [50]:
238990/total_urban_rural

0.3617339131408748

In [54]:
63.82 + 36.17


99.99000000000001

In [52]:
1 - 0.9999606465469616

3.935345303840432e-05

<h3>Insight 8: 63.82% of the accidents happen in Urban areas. 36.17% of the accidents happens in Rural areas. The remaining 0.01% are unaccounted for. This means that there are more accidents in places where there are more people present.</h3> 

<h2>Question: For the accidents in Urban areas, what are the top 5 vehicle types that causes the most accidents? What about the Rural areas?</h2>

In [69]:
accident['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      497992
Van / Goods 3.5 tonnes mgw or under       34160
Bus or coach (17 or more pass seats)      25878
Motorcycle over 500cc                     25657
Goods 7.5 tonnes mgw and over             17307
Motorcycle 125cc and under                15269
Taxi/Private hire car                     13294
Motorcycle over 125cc and up to 500cc      7656
Motorcycle 50cc and under                  7603
Goods over 3.5t. and under 7.5t            6096
Other vehicle                              5637
Minibus (8 - 16 passenger seats)           1976
Agricultural vehicle                       1947
Pedal cycle                                 197
Data missing or out of range                  6
Ridden horse                                  4
Name: count, dtype: int64

In [67]:
urban['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      316053
Van / Goods 3.5 tonnes mgw or under       22003
Bus or coach (17 or more pass seats)      16850
Motorcycle over 500cc                     16700
Goods 7.5 tonnes mgw and over             11151
Motorcycle 125cc and under                10246
Taxi/Private hire car                      8612
Motorcycle over 125cc and up to 500cc      4982
Motorcycle 50cc and under                  4893
Goods over 3.5t. and under 7.5t            3864
Other vehicle                              3643
Agricultural vehicle                       1272
Minibus (8 - 16 passenger seats)           1258
Pedal cycle                                 127
Data missing or out of range                  6
Ridden horse                                  3
Name: count, dtype: int64

In [68]:
rural['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      181922
Van / Goods 3.5 tonnes mgw or under       12156
Bus or coach (17 or more pass seats)       9025
Motorcycle over 500cc                      8957
Goods 7.5 tonnes mgw and over              6156
Motorcycle 125cc and under                 5023
Taxi/Private hire car                      4677
Motorcycle 50cc and under                  2710
Motorcycle over 125cc and up to 500cc      2674
Goods over 3.5t. and under 7.5t            2232
Other vehicle                              1994
Minibus (8 - 16 passenger seats)            718
Agricultural vehicle                        675
Pedal cycle                                  70
Ridden horse                                  1
Data missing or out of range                  0
Name: count, dtype: int64

<h3>Insight 9: The top 5 vehicle type that causes the most accident on Urban Areas are the following: Car, Van / Goods 3.5 tonnes mgw or under, Bus or coach (17 or more pass seats), Motorcycle over 500cc, and Goods 7.5 tonnes mgw and over.</h3>

<h3>Insight 10: The top 5 vehicle type that causes the most accident on Rural Areas are the following: Car, Van / Goods 3.5 tonnes mgw or under, Bus or coach (17 or more pass seats), Motorcycle over 500cc, and Goods 7.5 tonnes mgw and over.</h3>

<h3>Insight 11: Despite being whole different areas, Urban and Rural areas have the same leading type of vehicles in causing accidents. This might indicate that the vehicle type that is much largely produced can cause the most accidents. The most common type of vehicle in United Kingdom are cars and vans, and they are the leading vehicle types for accidents. </h3>

<h2>Question: Using the oldest data availlable in the dataframe, can you find the total percentage and exact amount of the fatalities?</h2>

In [94]:
accident_2019 = accident[accident['Year'] == 2019]
accident_2019['Accident_Severity'].value_counts()

Accident_Severity
Slight     155079
Serious     24322
Fatal        2714
Name: count, dtype: int64

In [95]:
accident_2019['Accident_Severity'].value_counts().sum()

182115

In [96]:
2714 /144419

0.01879254114763293

<h3>Insight 12: In year 2019, 1.87% of the accidents are fatal with a number 2714 occurences.</h3>

<h2>Question: What about using the most recent data availlable in the dataframe, can you find the total percentage and exact amount of the fatalities?</h2>

In [97]:
accident_2022 = accident[accident['Year'] == 2022]
accident_2022['Accident_Severity'].value_counts()

Accident_Severity
Slight     124093
Serious     18777
Fatal        1549
Name: count, dtype: int64

In [98]:
accident_2022['Accident_Severity'].value_counts().sum()

144419

In [99]:
1549 / 144419

0.010725735533413194

<h3>Insight 13: In year 2022, 1.07% of the accidents are fatal with a number 1549 occurences.</h3>

<h3>Insight 14: Based on the analysis about Fatal accidents. We found out that they have significantly less numbers than that of Slight and Serious accidents combined with only less than 2% occurences. In comparison between the year 2019 and 2022, there are less fatal accidents in the recent years compared to older years. </h3>

<h2>Question: Using the data from 2022, What months causes the most overall accidents? What about the average accidents per month?</h2>

In [110]:
accident_2022.groupby(['Accident_Severity','Month'], observed=False).size().unstack()

Month              January  February  March  April    May   June   July  \
Accident_Severity                                                         
Fatal                  104       117    129    115    147    138    121   
Serious               1139      1290   1606   1599   1757   1740   1765   
Slight                8724      9528  10606   9796  10468  10934  10767   

Month              August  September  October  November  December  
Accident_Severity                                                  
Fatal                 160        144      151       133        90  
Serious              1626       1791     1762      1631      1071  
Slight              10302      11025    11621     11858      8464

In [111]:
accident_2022.groupby(['Accident_Severity','Month'], observed=False).size().unstack().sum()

Month
January       9967
February     10935
March        12341
April        11510
May          12372
June         12812
July         12653
August       12088
September    12960
October      13534
November     13622
December      9625
dtype: int64

<h3>Insight 15: The month with the highest accident rate during 2022 is November, closely followed by October, with only a margin of 88 accidents. This occurence might be due to Halloween event every November.</h3>

<h3>Insight 16: The month with the lowest accident rate during 2022 is December, followed closely by January, and then relatively followed by February. These are the months where most people are usually on vacation.</h3>

<h2>Question: Since November and October is the month where there is the highest accidents. Does the Halloween have to do with this? Halloween is celebrated during October 31, for some people it extends to early November 1.</h2>

In [134]:
accident_2022.groupby(['Day','Month'], observed=False).size().unstack()

Month  January  February  March  April  May  June  July  August  September  \
Day                                                                          
1          276       631    571    480  354   446   436     318        460   
2          274       408    524    327  239   448   492     318        441   
3          252       447    341    321  272   473   401     376        484   
4          379       363    430    276  446   439   338     426        375   
5          360       472    472    231  366   398   429     377        314   
6          254       336    381    324  369   293   400     411        423   
7          263       213    352    349  441   410   382     374        468   
8          301       358    483    405  376   449   443     295        424   
9          236       433    341    438  270   429   487     388        460   
10         123       546    400    407  404   352   392     458        474   
11         263       417    383    304  402   449   343     370        429   
12         262       407    464    347  417   372   446     397        339   
13         376       331    372    426  403   312   461     495        373   
14         278       276    297    343  475   385   466     378        465   
15         345       376    353    383  422   470   442     320        433   
16         269       395    388    428  306   488   455     400        431   
17         300       424    364    413  407   452   342     404        486   
18         383       367    406    313  434   472   278     384        412   
19         351       419    457    351  364   339   420     430        322   
20         356       412    370    418  405   346   396     425        462   
21         389       334    305    430  490   439   487     374        442   
22         401       403    377    432  468   518   412     346        484   
23         287       357    402    501  442   506   449     427        513   
24         232       390    390    389  489   452   345     397        445   
25         325       392    422    291  436   493   272     435        408   
26         362       448    468    420  409   449   399     470        317   
27         381       328    346    455  476   354   359     404        441   
28         383       252    301    392  523   482   437     391        393   
29         479         0    400    422  478   466   405     335        561   
30         515         0    403    494  326   431   440     328        481   
31         312         0    378      0  263     0   399     437          0   

Month  October  November  December  
Day                                 
1          656       432       245  
2          463       457       236  
3          346       520       288  
4          487       489       255  
5          411       598       326  
6          477       396       412  
7          455       317       355  
8          482       530       390  
9          360       498       444  
10         367       517       412  
11         469       472       378  
12         486       549       303  
13         406       374       455  
14         402       327       444  
15         482       568       441  
16         410       537       423  
17         329       450       540  
18         399       541       274  
19         525       584       211  
20         508       375       301  
21         452       273       235  
22         464       433       347  
23         427       472       307  
24         313       460       325  
25         448       451       145  
26         470       544       195  
27         439       415       179  
28         366       274       181  
29         420       377       214  
30         471       392       171  
31         344         0       193

<h3>Insight 17: In October 31, There are 344 accidents. In November 1, there 432 accidents. The Day in October and November which has the highest accidents is October 1 and November 5. This means that Halloween event does not heighten the amount of accidents. But, it does lower the accident rate for October 31, the start of Halloween.

In [135]:
accident

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious    2019-06-05  51.506187   
1       200701BS65737           Serious    2019-07-02  51.495029   
2       200701BS66127           Serious    2019-08-26  51.517715   
3       200701BS66128           Serious    2019-08-16  51.495478   
4       200701BS66837            Slight    2019-09-03  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    2022-02-18  57.374005   
660675  201091NM01881            Slight    2022-02-21  57.232273   
660676  201091NM01935            Slight    2022-02-23  57.585044   
660677  201091NM01964           Serious    2022-02-23  57.214898   
660678  201091NM02142           Serious    2022-02-28  57.575210   

                   Light_Conditions           District Area Longitude  \
0             Darkness - lights lit  Kensington and Chelsea -0.209082   
1                          Daylight  Kensington and Chelsea -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea -0.210215   
3                          Daylight  Kensington and Chelsea -0.202731   
4             Darkness - lights lit  Kensington and Chelsea -0.192487   
...                             ...                     ...       ...   
660674                     Daylight                Highland -3.467828   
660675       Darkness - no lighting                Highland -3.809281   
660676                     Daylight                Highland -3.862727   
660677       Darkness - no lighting                Highland -3.823997   
660678                     Daylight                Highland -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                    Road_Type Urban_or_Rural_Area  \
0          Single carriageway               Urban   
1          Single carriageway               Urban   
2       Unaccounted Road Type               Urban   
3          Single carriageway               Urban   
4       Unaccounted Road Type               Urban   
...                       ...                 ...   
660674     Single carriageway               Rural   
660675     Single carriageway               Rural   
660676     Single carriageway               Rural   
660677     Single carriageway               Rural   
660678       Dual carriageway               Rural   

                    Weather_Conditions                          Vehicle_Type  \
0                   Fine no high winds                                   Car   
1                Raining no high winds                                   Car   
2       Unaccounted Weather Cobditions                 Taxi/Private hire car   
3                   Fine no high winds  Bus or coach (17 or more pass seats)   
4       Unaccounted Weather Cobditions                         Other vehicle   
...                                ...                                   ...   
660674              Fine no high winds                                   Car   
660675              Fine no high winds                                   Car   
660676              Fine no high winds              

<h2>Question: Using the recent years from 2020 to 2022, Which District Area is most prone to accidents? </h2>

In [152]:
accident_2019 = accident[accident['Year'] == 2019]
accident_2020 = accident[accident['Year'] == 2020]
accident_2021 = accident[accident['Year'] == 2021]
accident_2022 = accident[accident['Year'] == 2022]

In [156]:
accident_2020['District Area'].value_counts()

District Area
Birmingham                   3506
Leeds                        2333
Manchester                   1756
Glasgow City                 1651
Bradford                     1570
                             ... 
Cheshire East                   0
Northumberland                  0
Central Bedfordshire            0
Cheshire West and Chester       0
Cornwall                        0
Name: count, Length: 422, dtype: int64

In [154]:
accident_2021['District Area'].value_counts()

District Area
Birmingham            3289
Leeds                 2186
Manchester            1707
Bradford              1567
Glasgow City          1509
                      ... 
Chester-le-Street       25
Alnwick                 20
South Shropshire        18
Teesdale                15
Berwick-upon-Tweed      13
Name: count, Length: 422, dtype: int64

In [155]:
accident_2022['District Area'].value_counts()

District Area
Birmingham          2876
Leeds               1954
Bradford            1439
Manchester          1425
Westminster         1421
                    ... 
Aberdeenshire          0
Mid Bedfordshire       0
Midlothian             0
Moray                  0
Aberdeen City          0
Name: count, Length: 422, dtype: int64

<h3>Insight 18: The district that is most prone to accidents is Birmingham, followed by Leeds, and Manchester.</h3>

<h3>Question: From the data frame, what is the Road Type that causes the most accidents?</h3>

In [211]:
roadtype = accident['Road_Type'].value_counts()

In [212]:
total = accident['Road_Type'].value_counts().sum()

In [219]:
accident.groupby(['Accident_Severity','Road_Type'], observed=False).size().unstack()

Road_Type          Dual carriageway  One way street  Roundabout  \
Accident_Severity                                                 
Fatal                          1815              95         142   
Serious                       11746            1655        3665   
Slight                        85863           11809       40185   

Road_Type          Single carriageway  Slip road  Unaccounted Road Type  
Accident_Severity                                                        
Fatal                            6527         49                     33  
Serious                         70059        611                    481  
Slight                         415557       6381                   4006

In [220]:
roadtype / total

Road_Type
Single carriageway       0.744905
Dual carriageway         0.150488
Roundabout               0.066586
One way street           0.020523
Slip road                0.010657
Unaccounted Road Type    0.006841
Name: count, dtype: float64

<h3>Insight 19: The road type that has the highest accident is Single carriageway. Single Carriageway causes 74% of the accidents.</h3>

<h2>Question: Which Day of the week has the most amount of accidents?</h2>

In [223]:
accident['DaysOfWeek'].value_counts()

DaysOfWeek
Saturday     107178
Wednesday     99558
Thursday      99511
Friday        97900
Tuesday       94550
Sunday        89302
Monday        72680
Name: count, dtype: int64

<h3>Insight 20: The day of the week that has the most accidents is Saturday. This might be because more people are going out to relax or spend time out of work. Causing them to travel and go outside which makes them prone to higher chances of accidents.
</h3>